Keywords in database

egedebruin opened the pull request:
This makes it able to add, delete en get tags to/from the database.

I will create tests for this if the other branch is merged and i can merge from develop. I maked this pull request so you can review it already.

Kingdorian commented:
I might be mistaken but I don't see any tests for the methods. I think you should add those before we can merge this.

egedebruin commented:
You are not mistaking, I will add those test when the other branch is merged into develop.

jochem725 commented:
@egedebruin Why will you add the tests when the other branch is merged into develop? Is there a dependency you need?

jochem725 commented on backend/src/main/java/com/benine/backend/database/MySQLDatabase.java:
```diff
@@ -245,6 +246,71 @@ public void useDatabase() throws SQLException {
+      e.printStackTrace();
+      getLogger().log("Tag couldn't be deleted.", LogEvent.Type.CRITICAL);
+    } finally {
+      if (statement != null) {
```
This is duplicate code which could be in a different method. Also, please don't print the stack trace but use the logger instead.

egedebruin commented:
I have added the tests for the new features i added, I think it is ready for merging now.

I also took you comment into consideration @jochem725 and fixed it in the whole MySQLDatabase class.

bvanwijk commented:
@egedebruin There are some findbugs error in MySQLDatabase maybe you can fix them before we merge.

egedebruin commented:
Thanks for noticing @bvanwijk , I have fixed them.

egedebruin commented:
I have added that it is possible to add tags to, delete tags from and get tags from the database and it will load the presets on startup.

And the needed tests of course (:

bvanwijk commented:
Good work Ege, nice that you've fixed all the errors I agree with merging.

jochem725 commented:
Great! I think this can be merged.
